util/multirun.sh

```diff
@@ -36,26 +68,99 @@ fi
 cip="$1"
 shift
 
-for opts in "$@"; do
-    manifest=$(echo "$opts" | cut -d, -f1)
-    service_accounts=()
-    service_account_keyfiles=()
-    while IFS= read -r keyfile; do
-        service_account_keyfiles+=( "$keyfile" )
-    done < <( echo "$opts" | cut -d, -f2- | tr , '\n' )
-    for keyfile in "${service_account_keyfiles[@]}"; do
-        # Authenticate as the service account. This allows the promoter to later
-        # call gcloud with the flag `--account=...`. We can allow the service
-        # account creds file to be empty, for testing cip locally (for the case
-        # where the service account creds are already activated).
-        gcloud auth activate-service-account --key-file="${keyfile}"
-        service_accounts+=("$(gcloud config get-value account)")
+for arg; do
+    if ! [[ "$arg" =~ .*,.* ]]; then
+        echo >&2 "invalid argument: $arg"
+        usage
+        exit 1
+    fi
+done
+
+args_final=("$@")
+# If we only want to run against changed manifest files, then filter out those
+# files that were not changed since the last commit.
+if [[ "${CIP_GIT_REV:-}" ]]; then
+    if [[ ! -d "${CIP_GIT_DIR:-}" ]]; then
+        echo >&2 "CIP_GIT_DIR not set (must be set if CIP_GIT_REV is set)"
+        usage
+        exit 1
+    fi
+
+    # Get a lit of all files that changed in the given CIP_GIT_REV.
+    # See https://stackoverflow.com/a/424142/437583.
+    changed_files=()
+    while IFS= read -r f; do
+        changed_files+=( "$f" )
+    done < <( git -C "${CIP_GIT_DIR}" diff-tree --no-commit-id --name-only -r "${CIP_GIT_REV}" )
+
+    # Filter those changed files to only have promoter manifests.
+    changed_manifests=$(collect_manifests "${cip}" "${changed_files[@]}")
```

Comment:
Hmm, I think I understand your concern. E.g., it may be the case that we have a valid manifest file, but it gets changed with an errant typo (making it unparseable) in the PR.

I think I can reorder the filtering logic to record an error if a given manifest file argument to multirun.sh fails to parse (this is a condition that should never happen, regardless of Git changed files). If they all parse correctly, then we should optimize by filtering out those files that have not changed.